libs/etest/src/etest.erl

```diff
@@ -119,8 +119,35 @@ assert_failure(F, E) ->
 
 %% @private
 run_test(Test) ->
+    Parent = self(),
+    {Pid, Ref} = spawn_opt(
+        fun() ->
+            Result = do_run_test(Test),
+            Parent ! {self(), Result}
+        end,
+        [monitor]
+    ),
+    receive
+        {Pid, Result} ->
+            receive
+                {'DOWN', Ref, process, Pid, normal} -> ok
+            after 0 -> ok
+            end,
+            Result;
+        {'DOWN', Ref, process, Pid, Reason} ->
+            {error, Reason}
+    end.
```

Comment:
I can't tell you how many abandoned branches I have done the same (to eliminate message pollution)

Reply:
As I'm breaking this PR into smaller pieces, I moved this to #793 where it is needed.